src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -727,21 +727,23 @@ mongoc_uri_option_is_int32 (const char *key)
           !strcasecmp (key, MONGOC_URI_HEARTBEATFREQUENCYMS) ||
           !strcasecmp (key, MONGOC_URI_SERVERSELECTIONTIMEOUTMS) ||
           !strcasecmp (key, MONGOC_URI_SOCKETCHECKINTERVALMS) ||
-          !strcasecmp (key, MONGOC_URI_SOCKETTIMEOUTMS) ||
           !strcasecmp (key, MONGOC_URI_LOCALTHRESHOLDMS) ||
           !strcasecmp (key, MONGOC_URI_MAXPOOLSIZE) ||
           !strcasecmp (key, MONGOC_URI_MAXSTALENESSSECONDS) ||
           !strcasecmp (key, MONGOC_URI_MINPOOLSIZE) ||
           !strcasecmp (key, MONGOC_URI_MAXIDLETIMEMS) ||
           !strcasecmp (key, MONGOC_URI_WAITQUEUEMULTIPLE) ||
+          !strcasecmp (key, MONGOC_URI_ZLIBCOMPRESSIONLEVEL) ||
+          /* deprecated options */
```

Comment:
If `MONGOC_URI_TIMEOUTMS` is mixed with a deprecated timeout option it will override the behavior of the deprecated option.

I suggest warning with `MONGOC_WARNING` if a URI contains `MONGOC_URI_WAITQUEUETIMEOUTMS` or `MONGOC_URI_SOCKETTIMEOUTMS`.

`mongoc_uri_apply_options` may be a good spot to add that check. That is a common to both ways options are set: options specified in the URI string or options retrieved from [TXT records](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst#default-connection-string-options). Though `timeoutMS` would not be applied through TXT records currently.

Reply:
Done.